website/content/en/status/report-2023-07-2023-09/valgrind.adoc

```diff
@@ -0,0 +1,22 @@
+=== Valgrind - valgrind-devel updated for FreeBSD 15
+  
+Links: +
+link:https://www.valgrind.org/[Valgrind Home Page] URL: link:https://www.valgrind.org/[] +
+link:https://www.valgrind.org/docs/manual/dist.news.html[Valgrind News] URL: link:https://www.valgrind.org/docs/manual/dist.news.html[]
+
+Contact: Paul Floyd <pjfloyd@wanadoo.fr>
+
+The package:devel/valgrind-devel[] is in the process of being updated.
+This contains most of what will be in the official release of Valgrind 3.22 due out in October.
+
+memcheck has been enhanced with some more checks. It will now report usage of realloc with a size of zero.
+Such usage is not portable and is deprecated (C23 will make it Undefined Behaviour). memcheck
+now validates the values used for alignment and sized delete for memalign, posix_memalign, aligned_alloc
+and all aligned and sized overloads of operator new and operator delete. Reading DWARF debuginfo is
+now done in a lazy manner which can improve performance.
+
+As usual there are numerous small bugfixes.
+
+Specific to FreeBSD there is support for FreeBSD 15. Two extra _umtx_op operations are now supported,
+UMTX_OP_GET_MIN_TIMEOUT and UMTX_OP_SET_MIN_TIMEOUT. There is a fix for the use of
+sysctl kern proc pathname with the guest pid or -1 which previously returned the path of the Valgrind host.
```

Comment:
```suggestion
Specific to FreeBSD there is support for FreeBSD 15. Two extra `_umtx_op` operations are now supported, `UMTX_OP_GET_MIN_TIMEOUT` and `UMTX_OP_SET_MIN_TIMEOUT`.
There is a fix for the use of sysctl `kern.proc.pathname` with the guest pid or -1, which previously returned the path of the Valgrind host.
```
Was _kern proc pathname_ (with spaces) intended?

```text
% sysctl -d kern.proc.pathname
kern.proc.pathname: Process executable path
% 
```

Whilst here: a comma.

Reply:
I'd rather leave it as it is. It's not a command line sysctl. Strictly speaking not even a libc sysctl.